### PR TITLE
Test snapshot restore with volume mode and access modes different than parent PVC

### PR DESCRIPTION
**Pull request by jilju**
Test snapshot restore with volume mode and access modes different than parent PVC

### CODE REVIEW

**Comment by jilju** [comment]
Rebased

**Comment by petr-balogh** [comment]
@jilju are you going to finish with this PR? Needs rebase.

File: tests/manage/pv_services/pvc_snapshot/test_restore_with_different_volume_mode.py [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+        )
+
+    def test_snapshot_restore_with_different_volume_mode(
+        self, pod_factory, snapshot_factory, snapshot_restore_factory
+    ):
+        """
+        Restore snapshot with volume mode - access mode combination different than the parent PVC
+
+        """
+        file_name = "fio_test"
```
Comment (petr-balogh) [inline tests/manage/pv_services/pvc_snapshot/test_restore_with_different_volume_mode.py:60]: Can there be some conflict in file name when running the test for second time?
The same for file_name_new.

Should we add some random characters to the file name?

**Review by petr-balogh — APPROVED** [review]

File: tests/manage/pv_services/pvc_snapshot/test_restore_with_different_volume_mode.py (hunk 1 above)
Comment (jilju) [reply tests/manage/pv_services/pvc_snapshot/test_restore_with_different_volume_mode.py:60]: Entirely new PVCs are pods ill be created when the test case is running for the second time. So no conflicts will arise.

**Pull request closed without merging**